app/models/impression.rb

```diff
@@ -37,6 +37,7 @@ class Impression < ApplicationRecord
 
   # callbacks .................................................................
   # scopes ....................................................................
+  scope :clicked, -> { where.not(clicked_at: nil) }
```

Comment:
We don't current have an index to optimize this query. How often will we ask for only clicks vs all impressions then determining clicks from that set? Is this something we should optimize for?

Reply:
probably